### PR TITLE
Customizer changes for customizing staking fields

### LINKED ISSUE #3049: Update `HederaAccountCustomizer` usage for staking fields
### Problem

The `HederaAccountCustomizer` encapsulates a set of changes to an account, and is used directly or indirectly by:

- [x] `CryptoCreateTransitionLogic`
- [x] `CryptoUpdateTransitionLogic`
- [x] `ContractUpdateTransitionLogic`
- [x] Contract operations via the [soon-to-be-merged `ContractCustomizer`](https://github.com/hashgraph/hedera-services/blob/03039-M-FixChildAdminKeyInheritance/hedera-node/src/main/java/com/hedera/services/ledger/accounts/ContractCustomizer.java#L43).

All of these usages should be updated to respect HAPI-configurable staking fields (`staked_node_id`/`staked_account_id`/`decline_reward`) and **ignore** the legacy `proxyAccountID` field.

### Solution

Update `HederaAccountCustomizer` usage for staking fields.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #3049 

- Updated `HederaAccountCustomizer` and `ContractCustomizerFactory` to ustomize `stakedId`, `declinedReward`
- Added changes to `getAccountInfo` and `getContractInfo` to display the staking fields. Only the `pending_paid_rewards` field will be populated in future PR
- Made changes to `CryptoCreateTransitionLogic`, `CryptoUpdateTransitionLogic`, `ContractCreateTransitionLogic` and `ContractUpdateTransitionLogic` to ignore legacy field `proxyAccountID` and validate/set new staking fields

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleAccount.java [hunk 1]
```diff
@@ -446,24 +446,29 @@ public void setStakePeriodStart(long stakePeriodStart) {
...
 
 	/**
-	 * Set the node to which this account is staking its Hbar to
+	 * Set the node to which this account is staking its Hbar to. The value of the nodeNum will be negative if its
+	 * staked node id.
+	 *
 	 * @param nodeNum
 	 * 		The node num of the node
 	 */
 	public void setStakedNodeId(long nodeNum) {
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleAccount.java:463]: Imo `stakedId` would be a clearer name, with the javadoc updated to note this is an account number when `stakedId > 0` and `-stakedId` is a node id when `stakedId < 0`.

File: hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoUpdateTransitionLogic.java [hunk 2]
```diff
@@ -221,7 +233,12 @@ private ResponseCodeEnum validate(TransactionBody cryptoUpdateTxn) {
 		if (op.hasAutoRenewPeriod() && !validator.isValidAutoRenewPeriod(op.getAutoRenewPeriod())) {
 			return AUTORENEW_DURATION_NOT_IN_RANGE;
 		}
-
+		if (op.hasProxyAccountID() && !op.getProxyAccountID().equals(AccountID.getDefaultInstance())) {
+			return PROXY_ACCOUNT_ID_FIELD_IS_DEPRECATED;
+		}
+		if (!validator.isValidStakedIdIfPresent(op.getStakedAccountId(), op.getStakedNodeId(), accounts.get())) {
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoUpdateTransitionLogic.java:239]: Imo this should also pass the `NodeInfo` singleton so we can validate the `nodeId`, if that option is chosen.

**Review by tinker-michaelj — COMMENTED** [review]
Looks great in general! Just one nit on a parameter name + a possible added validation. 😇

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleAccount.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleAccount.java]: Changed `setStakedAccountId` and `setStakedNodeId` to `setStakedId` and changed the parameter name.

File: hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoUpdateTransitionLogic.java (hunk 2 above)
Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoUpdateTransitionLogic.java]: Passed NodeInfo to the 4 transition logics and validated using `nodeInfo.accountOf(nodeId)`

File: hedera-node/src/main/java/com/hedera/services/ledger/accounts/HederaAccountCustomizer.java [hunk 3]
```diff
@@ -66,8 +75,16 @@ public void customizeSynthetic(final ContractCreateTransactionBody.Builder op) {
 			op.setAutoRenewPeriod(Duration.newBuilder()
 					.setSeconds((long) changes.get(AccountProperty.AUTO_RENEW_PERIOD)));
 		}
-		if (changes.containsKey(AccountProperty.PROXY)) {
-			op.setProxyAccountID(((EntityId) changes.get(AccountProperty.PROXY)).toGrpcAccountId());
+		if (changes.containsKey(AccountProperty.STAKED_ID)) {
+			final long id = (long) changes.get(AccountProperty.STAKED_ID);
+			if (id <= 0) {
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/ledger/accounts/HederaAccountCustomizer.java:80]: Should this be strictly `< 0`? (If I understood, `stakedId == 0` meant this account does not stake to anything.)

Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/ledger/accounts/HederaAccountCustomizer.java]: Ah yes .. 😞  Changed it . Thanks @tinker-michaelj

File: hedera-node/src/main/java/com/hedera/services/txns/validation/PureValidation.java [hunk 4]
```diff
@@ -130,4 +133,37 @@ public static ResponseCodeEnum checkKey(final Key key, final ResponseCodeEnum fa
...
+	 *
+	 * @param nodeId
+	 * 		given node id
+	 * @param nodeInfo
+	 * 		nodeInfo object
+	 * @return true if valid, false otherwise
+	 */
+	private static boolean isValidNodeId(final long nodeId, final NodeInfo nodeInfo) {
+		try {
+			nodeInfo.accountOf(nodeId);
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/txns/validation/PureValidation.java]: Got it ! Changed

**Review by tinker-michaelj — APPROVED** [review]
LGTM!!

**Pull request merged**